### ISSUE TITLE
TypeError when overriding export entries in star export

### ISSUE THREAD

**Issue by legendecas** [microsoft/TypeScript#38820]
**TypeScript Version:** 3.9.3

**Search Terms:** export star, export binding, duplicated exports

**Code**

```ts
// a.ts
export const name = 'a'
```

```ts
// b.ts
export const name = 'b'
```

```ts
// c.ts
export * from './a'
export { name } from './b'
```

```ts
// d.ts
import { name } from './c'
console.log(name) 
```

**Expected behavior:**

In `d.ts`, it should print `b`.

Tested with pure .mjs with Node.js, it works as expected.

**Actual behavior:**

Throws `TypeError: Cannot redefine property: name`

Generates:
```js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
exports.__esModule = true;
__exportStar(require("./a"), exports);
var b_1 = require("./b");
__createBinding(exports, b_1, "name");
```

Also tested on TypeScript 3.8.3, 3.8.3 generates: 

```js
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
exports.__esModule = true;
__export(require("./a"));
var b_1 = require("./b");
exports.name = b_1.name;
```

According to https://tc39.es/ecma262/#sec-resolveexport, the star exports should be resolved after explicit export names.

**Comment by patrickm** [comment]
Having the same issue

Can reproduce on:
Version 3.9.5
Version 4.0.0-dev.20200626

On 3.8.2 works as expected

Looks like it's broken since 3.9.X by `Exports Now Use Getters for Live Bindings`

**Comment by rbuckton** [comment]
I've tested this against the latest in `master` and it works as expected. I believe this was fixed by #39213. I'm not sure why it would repo in 4.0.0-dev.20200626, as the fix should have been included in that release.
